### PR TITLE
fix: add time format rules

### LINKED ISSUE #8754: Date field truncates information
### JabRef version

5.6 (latest release)

### Operating system

macOS

### Details on version and operating system

macOS Monterey Version 12.4 Beta

### Checked with the latest development build

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Create a new entry in the library
2. Edit the date field to be a valid ISO timestamp including a time component (e.g., `2022-02-23T17:43:16`)
3. Click into another field

When the focus changes in step 3, the date field is truncated to just the date part.

If the time component does not include seconds, #8747 is triggered instead, but the information is not truncated.

### Appendix

_No response_

### PR DESCRIPTION
**Pull request by Christina0114**
Refs #8754 
 ### JabRef version
5.4 (latest release)

### Operating system
Mac OS

### Details on version and operating system
Monterey 12.2.1

### Descripion
After checking the code, I find the final format shown to the user is "yyyy-mm-dd". If the user inputs "2018-10-03T07:24:14+03:00", the time information will be truncated automatically. It is an unavoidable result. 
However, if the user wants to input "2022-02-23T17:43", it shouldn't trigger an error. Based on this, I added some rules in the "Date" class.
            
"uuuu-MM-dd'T'HH:mm",   // covers 2018-10-03T07:24. 
                "uuuu-MM-dd'T'HH:m",   // covers 2018-10-03T17:2
                "uuuu-MM-dd'T'H:mm",   // covers 2018-10-03T7:24
                "uuuu-MM-dd'T'H:m",  // covers 2018-10-03T7:7
                "uuuu-MM-dd'T'HH",   // covers 2018-10-03T07
                "uuuu-MM-dd'T'H",    // covers 2018-10-03T7

_after fix:_
when you input 2018-10-03T17:2 or 2018-10-03T17:22 or 2018-10-03T2:3 in the "Date", the result is below:
Screen Shot 2022-05-10 at 16 31 53

- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [x] Screenshots added in PR description (for UI changes)
- [x] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.

### CODE REVIEW

File: src/main/java/org/jabref/model/entry/Date.java [hunk 1]
```diff
@@ -21,6 +21,18 @@ public class Date {
     static {
         List<String> formatStrings = Arrays.asList(
                 "uuuu-MM-dd'T'HH:mm:ss[xxx][xx][X]",    // covers 2018-10-03T07:24:14+03:00
+                /*
+                 start: fix 8754
+                 */
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/model/entry/Date.java:26]: Please remove that comment, unnecessary. Please also add some tests for the new format

**Comment by calixtus** [comment]
I took a look into the code and realized that there is a deeper issue with the DateTimeParser. It seems that the first pattern including the timezone is somehow not parsed correctly. I'll continue tomorrow in investigation.

**Comment by Christina0114** [comment]
>

**Comment by Christina0114** [comment]
> I took a look into the code and realized that there is a deeper issue with the DateTimeParser. It seems that the first pattern including the timezone is somehow not parsed correctly. I'll continue tomorrow in investigation.

**Comment by Christina0114** [comment]
sorry, I closed it by accident. Which rule do you refer to?

**Comment by Siedlerchr** [comment]
The failing unit test with the TimeZone comes from the fact that we just use a LocalDateTime Object which is by default not timezone aware. 
https://mkyong.com/java8/java-8-zoneddatetime-examples/

**Comment by calixtus** [comment]
It was a stupid mistake on my side. As I was working on the test cases and a fix for the toString method, I altered your parsing patterns and put your parsing pattern meanwhile in a comment. Then I was wondering why it did not parse the zone-offset.
kudos @Siedlerchr for putting me on the right track.

All relevant tests are green, ready to review and merge

**Comment by calixtus** [comment]
This PR does not fix the issue in #8754 since this seems to be a problem of the TemporalAccessorPicker or the DateEditorViewModel. But: the patterns of @Christina0114 are still a valuable addition. So I'd like to include this PR anyway.

**Pull request merged**